### PR TITLE
Docs warning for charge in DielectricConstant

### LINKED ISSUE #4262: [Docs] document limits of DielectricConstant
Follow-up from https://github.com/MDAnalysis/mdanalysis/pull/4249#issuecomment-1696077930

As per: https://github.com/MDAnalysis/mdanalysis/pull/4249#issuecomment-1696077930

The limits of DielectricConstant, especially regarding free charge, should be better documented.

### PR DESCRIPTION
**Pull request by PicoCentauri**
Fixes #4262

Add a warning box to the documentation that only charge neutral systems can be handled by the current implementation.
See also #4249 for details on why this is important.

PR Checklist
------------
 - [x] Tests (we have a test for the error raise already.)?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
:books: Documentation preview :books:: https://mdanalysis--4263.org.readthedocs.build/en/4263/

### CODE REVIEW

File: package/MDAnalysis/analysis/dielectric.py
Comment (IAlibay) [inline package/MDAnalysis/analysis/dielectric.py:69]: Maybe a stupid question, if this is relying on an Ewald style summation, how well does it handle a system with a net charge?

Comment (PicoCentauri) [reply package/MDAnalysis/analysis/dielectric.py:69]: No, totally valid question. The answer will be a bit longer. sorry.

Ewald can handle systems with a net charge. You introduce a homogenous background charge that neutralizes the system, which will lead to an additional force an all atoms. This is unproblematic for homogenous systems but [lead to severe artifacts in inhomogenous systems like membranes etc](http://dx.doi.org/10.1021/ct400626b).

The equation we use here is derived for an Ewald boundary condition, meaning that the dielectric constant of sorrounding medium is a metal ($\varepsilon=\infty$).  If you have other boundary conditions like when using reaction field or cutoff electrostatics the equation will change as described in the paper we cite. We use eq 26.4.3D because the derivation for an Ewald electrostatic leads to the same results as a reaction field with an dielectric constant of $\infty$.

Screenshot 2023-08-28 at 21 39 54

Now, why are charges a problem? The issue is that in the derivation you demand that Maxwell's equation for electric field

$$
\nabla \vec E = \frac{\rho}{\varepsilon_0}
$$

where $\rho$ is the charge density is equal to zero: $\nabla \vec E = 0$. This means that no free charges are present. If you do not do this you will not end up with the equation we use.

Does this answer your question?

Comment (hmacdope) [reply package/MDAnalysis/analysis/dielectric.py:69]: Do you think you could transfer this very insightful piece of writing to an issue for future reference. :)

File: package/MDAnalysis/analysis/dielectric.py
Comment (hmacdope) [inline package/MDAnalysis/analysis/dielectric.py:70]: @PicoCentauri  perhaps I am misunderstanding but you can have no  charged anything or is it that you need overall neutrality. Apologoies if silly q.

File: package/MDAnalysis/analysis/dielectric.py
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/dielectric.py:69]: Yes we sure. If people find it there.

File: package/MDAnalysis/analysis/dielectric.py
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/dielectric.py:70]: Nono it is fine. This is really a bit tricky. There are even publications where people do this wrong.

As we do it in the assertion in the class' prepare function you need charge neutral molecules. In a classical MD world they can have of course have partial charges. In fact, they have to because otherwise the system's total dipole moment would be zero and with it the dielectric constant.

What you can not have are ions such as Na+ Cl- or other charged fragments i.e negative carboxylic acid groups COO- or positive amine groups H3N+.

**Review by hmacdope — APPROVED** [review]
Im happy with the warning as is, once we resolve conflicts I think we can merge and address better documentation in #4262.

**Review by IAlibay — APPROVED** [review]
LGTM. Apologies I jumped in here and fixed the changelog conflicts. (no need to wait for CI again, this is a docs change anyways)

**Pull request merged**
